source/material_model/steinberger.cc

```diff
@@ -465,6 +517,10 @@ namespace aspect
           options.list_of_allowed_keys = compositional_field_names;
           viscosity_prefactors = Utilities::MapParsing::parse_map_to_double_array(prm.get("Composition viscosity prefactors"), options);
 
+          // Drucker-Prager rheology related
+          angle_of_internal_friction = prm.get_double ("Angle of internal friction") * constants::degree_to_radians;
+          cohesion = prm.get_double("Cohesion");
+
```

Comment:
Can you replace this by
```suggestion
          // Drucker-Prager rheology related
          enable_drucker_prager_rheology = prm.get_bool ("Use Drucker-Prager rheology");
          drucker_prager_plasticity.initialize_simulator (this->get_simulator());

          // drucker_prager_plasticity checks that n_phases has as many entries as the number of
          // chemical compositions, however we do not support different rheologies for different
          // compositions in this material model. Make sure the vector has the expected size, but
          // also notify drucker_prager_plasticity that there should be no values for other compositions.
          std::vector<unsigned int> n_phases = phase_function->n_phases_for_each_chemical_composition();
          for (unsigned int i=1; i<n_phases.size(); ++i)
            n_phases[i] = 0;
          drucker_prager_plasticity.parse_parameters(prm, std::make_unique<std::vector<unsigned int>> (n_phases));
```